janitor/utils.py

```diff
@@ -405,3 +406,227 @@ def _wrapped(x, *args, **kwargs):
+    are pretty clear from the code.
+    """
+    # dictionary should not be empty
+    if not entry:
+        raise ValueError("passed dictionary cannot be empty")
+    # If it is a NoneType, number, Boolean, or string,
+    # then wrap in a list
+    entry = {
+        key: [value]
+        if isinstance(value, (type(None), int, float, bool, str))
+        else value
+        for key, value in entry.items()
+    }
+
+    # Convert to list if value is a set|tuple|range
+    entry = {
+        key: list(value) if isinstance(value, (set, tuple, range)) else value
+        for key, value in entry.items()
+    }
+
+    # collect dataframes here
+    dfs = []
+
+    # collect non dataframes here, proper dicts
+    dicts = {}
+
+    for key, value in entry.items():
+
+        # exclude dicts:
+        if isinstance(value, dict):
+            raise TypeError("Nested dictionaries are not allowed")
+
+        # process arrays
+        if isinstance(value, np.ndarray):
+            if value.size == 0:
+                raise ValueError("array cannot be empty")
+            elif value.ndim == 1:
+                dfs.append(pd.DataFrame(value, columns=[key]))
+            elif value.ndim == 2:
+                dfs.append(pd.DataFrame(value).add_prefix(f"{key}_"))
+            else:
+                raise TypeError(
+                    "expand_grid works with only vector and matrix arrays"
+                )
+        # process series
+        if isinstance(value, pd.Series):
+            if value.empty:
+                raise ValueError("passed Series cannot be empty")
+            if not isinstance(value.index, pd.MultiIndex):
+                # this section checks if the Series has a name or not
+                # and uses that information to create a new column name
+                # for the resulting Dataframe
+                if value.name:
+                    value = value.to_frame(name=f"{key}_{value.name}")
+                    dfs.append(value)
+                else:
+                    value = value.to_frame(name=f"{key}")
+                    dfs.append(value)
+            else:
+                raise TypeError("expand_grid does not work with pd.MultiIndex")
+        # process dataframe
+        if isinstance(value, pd.DataFrame):
+            if value.empty:
+                raise ValueError("passed DataFrame cannot be empty")
+            if not (
+                isinstance(value.index, pd.MultiIndex)
+                or isinstance(value.columns, pd.MultiIndex)
+            ):
+                # add key to dataframe columns
+                value = value.add_prefix(f"{key}_")
+                dfs.append(value)
+            else:
+                raise TypeError("expand_grid does not work with pd.MultiIndex")
+        # process lists
+        if isinstance(value, list):
+            if not value:
+                raise ValueError("passed Sequence cannot be empty")
+            elif np.array(value).ndim == 1:
```

Comment:
Following DeepSource's recommendations:

```suggestion
            if np.array(value).ndim == 1:
```

Reference: https://deepsource.io/gh/ericmjl/pyjanitor/run/a1eee997-ac46-451d-b916-3393713ba9eb/python/PYL-R1720